core/src/main/java/org/apache/accumulo/core/lock/ServiceLock.java

```diff
@@ -653,7 +637,7 @@ public synchronized void process(WatchedEvent event) {
 
   public static boolean isLockHeld(ZooCache zc, LockID lid) {
 
-    var zLockPath = path(lid.path);
+    var zLockPath = ServiceLockPaths.parse(Optional.empty(), lid.path);
```

Comment:
MetadataConstraints calls this method when checking the lock column. From what I can tell, `lid.path` is set to `context.getZooKeeperRoot()`, which will throw an error. Things seem to be working though....

Reply:
I ran this down. The `lid.path` is the full path of the zookeeper node. When the LockID is serialized into the Value of the LOCK_COLUMN in the metadata table, the zookeeper root is removed from the front of the string. When the LockID is deserialized from the Value, the zookeeper root is added back to the front of the string. This isn't an issue.